src/components/select/select.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { type ReactElement, useId } from "react"
+
+import * as SelectPrimitive from "@radix-ui/react-select"
+import clsx from "clsx"
+
+import s from "./select.module.scss"
+
+import { Typography } from "../typography"
+
+type Options = {
+  icon?: string
+  text: string
+  value: string
+}
+
+type Props = {
+  className?: string
+  id?: string
+  labelText?: ReactElement
+  minWidth?: number
+  name: string
+  onValueChange?: (items: string) => void
+  options: Options[]
+  placeholder?: string
+} & React.ComponentPropsWithoutRef<typeof SelectPrimitive.Root>
+
+export const Select = React.forwardRef<React.ElementRef<typeof SelectPrimitive.Root>, Props>(
+  (props, ref) => {
+    const {
+      children,
+      className,
+      disabled,
+      id,
+      labelText,
+      minWidth,
+      name,
+      onValueChange,
+      options,
+      placeholder,
+      ...restProps
+    } = props
+    // Generate unique ID if it is not provided
+    const generatedId = useId()
+    const selectId = id ?? generatedId
+
+    const items = options?.map(option => {
+      return (
+        <SelectPrimitive.Item className={s.item} key={option.value} value={option.value}>
+          <SelectPrimitive.ItemText>
+            {option.icon && <img alt={option.text} src={option.icon} />}
+            {option.text}
+          </SelectPrimitive.ItemText>
+        </SelectPrimitive.Item>
+      )
+    })
+
+    return (
+      <SelectPrimitive.Root
+        disabled={disabled}
+        name={name}
+        onValueChange={onValueChange}
+        {...restProps}
+      >
+        <Typography
+          as={"label"}
+          className={clsx(s.label)}
+          htmlFor={selectId}
+          variant={"regular_text_14"}
+        >
+          {labelText}
+          <SelectPrimitive.Trigger
+            className={clsx(s.trigger, className && className)}
```

Comment:
className={clsx(s.trigger, className)}
If className is undefined it is ignored

Reply:
fix